src/api/java/com/minecolonies/api/colony/buildings/modules/IBuildingModuleView.java

```diff
@@ -21,4 +22,22 @@
      * @return this module itself.
      */
     IBuildingModuleView setBuildingView(final IBuildingView buildingView);
+
+    /**
+     * Get the matching window for the module.
+     * @return the window.
+     */
+    Window getWindow();
```

Comment:
You will need an @OnlyIn here or on the view.....
Or optionally a Supplier of a Supplier structure....

This won't work on a dedicated server, or at least is very dangerous.

Reply:
Aren't views always only client side?

Reply:
might be not due to the way we're constructing buildings now in our buildings entry right

Reply:
It'd probably be a good idea to have client-side-only modules in a different package from the server-side-only modules, at least. It's easier to spot when something is improperly importing across sides when they're segregated.

Reply:
We're only giving it a producer in the entry, not an actual view. But I agree with @uecasm I'll make it a seperate folder

Reply:
I double checked, this works fine in SMP too. But I moved it into seperate folders